### ISSUE TITLE
R script with ESS cannot plot graphics

### ISSUE THREAD

**Issue by pietrodito** [syl20bnr/spacemacs#13282]
#### Description :octocat:
R script with ESS cannot plot graphics 

#### Reproduction guide :beetle:
- Start Emacs
- Launch R process with `M-x R`
- In R process buffer : `plot(0, 0)`

*Observed behaviour:* :eyes: :broken_heart:
```
Error in .External2(C_X11, d$display, d$width, d$height, d$pointsize,  : 
  unable to start device X11cairo
In addition: Warning message:
In (function (display = "", width, height, pointsize, gamma, bg,  :
  unable to open connection to X11 display ''
```

*Expected behaviour:* :heart: :smile:
The plot in a new X window.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. d44522ee1)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm ess emacs-lisp)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2

#### Backtrace :paw_prints:
```
No message in *Backtrace* with debug-on-error
```
It is noteworthly that those steps work perfectly in terminal but not in GUI.

**Comment by pietrodito** [comment]
I think the issue is not spacemacs related but a problem between R and Cairo... I will dig that...

**Comment by alex-bochkarev** [comment]
I experienced exactly the same bug on GNU/Linux with Emacs `28.0.50`, Spacemacs `v.0.300.0`, develop branch `rev. c4ad14a73`. Everything works if I run R from the terminal, but complaints about `X11cairo` from Emacs.

It seems that something breaks in R system when I start it from Emacs. What I ended up doing after some googling is to execute `Sys.setenv("DISPLAY"=":0.0")` every time I start ESS. That is, as per [ESS docs](https://ess.r-project.org/Manual/ess.html#Customizing-startup), put it into some `~/autorun.R` file and add custom init to my `.spacemacs`: 
```elisp
(add-hook 'ess-r-post-run-hook (lambda () (ess-load-file "~/autorun.R")))
```
I would appreciate if anyone suggested a more elegant solution (or hinted how would I figure one myself...).

**Comment by lebensterben** [comment]
@alex-bochkarev 
Thanks for the pointer to ESS docs.
I've made a PR to address this issue. Would you checkout that branch and test whether it fixes the problem?

**Comment by alexey0308** [comment]
@alex-bochkarev @pietrodito Could you please check the value of the variable DISPLAY in your .spacemacs.env files (`SPC f e e`)?

**Comment by alexey0308** [comment]
@lebensterben @smile13241324 I get now messages about sourcing temp files every time I start ESS, so I found this issue. Would you consider setting or cleaning environment variables file by a user to be a better solution, than creating and sourcing a file on every start? It looks like a machine-specific issue and no feedback from the issue creator was given regarding this PR.

**Comment by lebensterben** [comment]
@alexey0308 
The commit authored by me was just adopting the instruction in the ESS documentation.

What's the message you see? Is that distracting?

**Comment by smile13241324** [comment]
I see what you mean Alexey. Sourcing this file every time may not be the best solution i.e. on my Linux box emacs runs on Display 1:0 however currently this is overwritten with 0:0 in the R session causing the original issue again.

I have changed this now to only source this file if `DISPLAY` is not set already, in this case it should be save to assume DISPLAY=:0.0.

Please have a try and see if this works for you as well @alexey0308 @lebensterben.

**Comment by alexey0308** [comment]
Hi Maximilian, thank you for your changes.

I think, that the whole PR related code must be reverted, because it is not responsibility of spacemacs to take care about ESS functionality with regard of environment.

1) R has other ways to specify its configuration: setting variable in `.Renviron` file as `DISPLAY=:0`
 or running `Sys.setenv("DISPLAY"=":0")` in `.Rprofile` file. I guess, it is not universal, and for WSL or VM users it is not just ":0". 
2) even if one would have to write any ELISP code, it should be done in the upstream, i.e. in ESS:
a user running vanilla Emacs would face the same issue.

I think it is better not to introduce additional code, which one will have to support. 

In any case, thank you both for your contributions.

**Comment by lebensterben** [comment]
I kinda agree to revert that commit.

We can add a section in layer documentation about this issue though in case anyone needs it.

**Comment by alex-bochkarev** [comment]
> @alex-bochkarev @pietrodito Could you please check the value of the variable DISPLAY in your .spacemacs.env files (`SPC f e e`)?

Wow, thanks! It seems I was unaware of a basic feature. This solved an issue for me (and is the kind of solution I hoped to exist). So: 
- my `.spacemacs.env` indeed contained line `DISPLAY=:1`. Not sure when, and where the value came from, but this seemed to cause problems in several places (including ESS, in Python in a similar situation, couple of things in orgmode, etc.). 
- just regenerating `.spacemacs.env` with `spacemacs/force-init-spacemacs-env`, bound to `SPC f e C-e` in my case, solved all of these (changing the value to `:0.0` manually would also help, but this is not universal).

I also think at least in my case it was a broader "problem" of setting environment variables for Emacs, which  already has a good solution with `force-init-spacemacs-env`. I assume I somehow initiated the file update in a bad moment. So it seems indeed pretty machine-specific. (However, I did not quite get it from the [docs](https://develop.spacemacs.org/doc/DOCUMENTATION.html#environment-variables-and-path) how often / at which moments the `.env` file is updated, by default).

Anyways: I would agree now that re-setting variables in R (and python, and org...), like I did, was just suboptimal.

Thanks a lot, it does help!
